File: MeatChat/MCPostCell.m
//
//  NMeatCell.m
//  MeatChat
//
//  Created by Marcus Ramberg on 14.01.14.
//  Copyright (c) 2014 Nordaaker AS. All rights reserved.
//

#import "MCPostCell.h"

@implementation MCPostCell

- (id)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier
{
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
        // Initialization code
    }
    return self;
}

-(void)awakeFromNib
{
      self.videoPlayer=[[AVPlayer alloc] init];
  AVPlayerLayer *layer=[AVPlayerLayer playerLayerWithPlayer: self.videoPlayer];
  self.videoPlayer.actionAtItemEnd=AVPlayerActionAtItemEndNone;
  layer.frame=CGRectMake(0, 0, 100, 75);
  layer.videoGravity=AVLayerVideoGravityResizeAspectFill;
  [self.video.layer addSublayer: layer];
}

- (void)play
{
  NSLog(@"Tapped");
  [self.videoPlayer play];
}


- (void)setSelected:(BOOL)selected animated:(BOOL)animated
{
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

- (void)prepareForReuse
{
  [[NSNotificationCenter defaultCenter] removeObserver:self];
}

- (void)dealloc
{
  [[NSNotificationCenter defaultCenter] removeObserver:self];
}

- (void)playerItemDidReachEnd:(NSNotification *)notification
{
  AVPlayerItem *p = [notification object];
  [p seekToTime:kCMTimeZero];
}



@end

File: MeatChat/MCPostCell.swift
//
//  MCPostCell.swift
//  MeatChat2
//
//  Created by Marcus Ramberg on 22/03/16.
//  Copyright © 2016 Nordaaker AS. All rights reserved.
//

import UIKit
import AVFoundation
import AVFoundation.AVPlayer
import AVKit

class MCPostCell : UITableViewCell {
    @IBOutlet weak var video :UIView!
    @IBOutlet weak var timeLabel :UILabel!
    @IBOutlet weak var textView :UITextView!
    @IBOutlet weak var blockButton :UIButton!

    var videoPlayer :AVPlayer?

    
    override func awakeFromNib() {
        self.videoPlayer=AVPlayer();
        let layer:AVPlayerLayer=AVPlayerLayer(player: self.videoPlayer)
        self.videoPlayer!.actionAtItemEnd=AVPlayerActionAtItemEnd.none
        layer.frame=CGRect(x: 0, y: 0, width: 100, height: 75);
        layer.videoGravity=AVLayerVideoGravityResizeAspectFill;
        self.video?.layer.addSublayer(layer)
    }
    
    func playerItemDidReachEnd(_ notification:Notification)
    {
        let p:AVPlayerItem = notification.object as! AVPlayerItem
        p.seek(to: kCMTimeZero)
    }

}
